lib/tooling/clang-tidy-tool.js

```diff
@@ -53,7 +56,7 @@ class ClangTidyTool extends BaseTool {
         return fs.writeFile(
             path.join(dir, "compile_flags.txt"),
             compileFlags.join("\n")
-        ).then(() => super.runTool(sourcefile, args)
+        ).then(() => super.runTool(compilationInfo, sourcefile, args)
```

Comment:
If it's useful while doing this to move to `async`/`await` feel free. I realise it might add more mud in the water during other changes but thought I'd mention it in case it simplifies anything you're doing.

Reply:
atm async await is very much magic to me, but might take a look